chat-sdk-core/src/main/java/co/chatsdk/core/base/BaseLocationAdapter.java

```diff
@@ -0,0 +1,104 @@
+package co.chatsdk.core.base;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.os.Looper;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+
+import java.util.List;
+
+import androidx.core.content.ContextCompat;
+import co.chatsdk.core.session.ChatSDK;
+import co.chatsdk.core.utils.DisposableList;
+import co.chatsdk.core.utils.PermissionRequestHandler;
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+
+/**
+ * Created by Pepe on 01/25/19.
+ */
+
+public class BaseLocationAdapter {
+
+    protected final FusedLocationProviderClient locationClient;
+    protected final DisposableList disposableList = new DisposableList();
+
+    protected Context context() {
+        return ChatSDK.shared().context();
+    }
+
+    public BaseLocationAdapter() {
+        locationClient = LocationServices.getFusedLocationProviderClient(context());
+    }
+
+    public Observable<Location> requestLocationUpdates(long interval, int distance) {
+        return requestLocationUpdates(interval)
+                .distinctUntilChanged((l1, l2) -> l1.distanceTo(l2) < distance);
+    }
+
+    public Observable<Location> requestLocationUpdates(long interval) {
+        LocationRequest locationRequest = new LocationRequest();
```

Comment:
All the code needs to be inside the observable. 

```
return Observable.create(
```

Should be the first line of this function. Part of the way RxJava works is that code is only executed when `subscribe` is called. Because of that, any function that returns an observable should only have code inside the observable block.